### ISSUE TITLE
Support for additional headers

### ISSUE THREAD

**Issue by softprops** [stepancheg/grpc-rust#17]
At the moment it looks like call_impl supports a fixed set of headers when starting a request. The wire format outlines a few others http://www.grpc.io/docs/guides/wire.html.  in particular I'm wondering there is a way to start a request with the authorization header to authenticate requests? If not, would you be open to a pull request adding that support?

**Comment by stepancheg** [comment]
Yes, pull request would be great. But before starting implementation, let's discuss, what API for specifying additional header should look like.

**Comment by softprops** [comment]
Sry for the slow reply, I'm now swinging back in to see the state of things.

so looking that [this example](http://www.grpc.io/docs/guides/wire.html#example) for instance you may want to provide an interface for user defined headers I was digging a while back and found [this vec](https://github.com/stepancheg/grpc-rust/blob/3712032949dcf532cb9663e23f869ac531cfeaf4/src/client.rs#L77-L82) of the required ':'-prefixed headers. For the non `:`-prefixed headers, my first through would be to provide those as vec/hashset of `httpbis::Header`s.

I'm specifically interested in ( and blocked by ) send the authorization header. I took a look at [go grpc's auth docs](https://github.com/grpc/grpc-go/blob/5e3de3f21767d4e0e8dd0ee8383bbbfe172d35de/Documentation/grpc-auth-support.md)

After reading their strategy I changed my mind a bit. It's not really a thing I want to have to provide on each request but configuration I'd want to set once. I dug around a bit more and identified [GrpcClientConf](https://github.com/stepancheg/grpc-rust/blob/3712032949dcf532cb9663e23f869ac531cfeaf4/src/client.rs#L25-L27) which may actually be a nicer candidate for users to configure things default headers like authorization and timeouts. If you capture and add the conf data to the [GrpcClient](https://github.com/stepancheg/grpc-rust/blob/3712032949dcf532cb9663e23f869ac531cfeaf4/src/client.rs#L46-L48) instance, the `call_impl` could just append `self.conf.headers` to the list of headers it builds up for the request.

I like this best because the [bootstrapping of the client](https://github.com/stepancheg/grpc-rust/blob/3712032949dcf532cb9663e23f869ac531cfeaf4/grpc-examples/src/bin/greeter_client.rs#L12) tends to be where you want to configure authorization credentials and other details.

There's a separate question about structured Headers. I haven't looked into specific headers have defined value types. But I think it would be simplest to expose your `httpbis::Header` type as the interface for providing headers
